### DISCUSSION TITLE
Exclusion of training points

### DISCUSSION THREAD

**Discussion by JinglaiZheng** [lululxvi/deepxde#1513] [Q&A]
Dear Dr. Lu

First of all thank you for this perfect library deepxde.

  I have encountered two problems in calculating time-dependent partial differential equations in two-dimensional space domain, and I ask for your help here:

  After undimensionalization and scaling, my space domain is a simple square with two vertices (0,0) and (1,1), my time domain is (0,1),  I have Dirichlet boundary conditions u=0 on the left and lower boundary of this square, and Dirichlet boundary conditions u=1 on the right and upper boundary of this square. However, no matter how hard I try to adjust the hyperparameters, the total loss after long-term training is still very high, not less than 0.01. By using model.predict(input,operator=pde), I find that the residual difference at the junction of (0,1) and (1,0) is very large. Here two unequal Dirichlet boundary conditions are satisfied. There are two questions :
  1. How can I remove these spots during training? I tried to use exclusion to remove a certain amount of intersections through the code shown below, but the program took a long time before compile, so this idea failed. 
  2.  What is calculated by model.predict(operator=pde) without the analytic solution? My graph has a maximum of 1500, I've read the documentation, but I still don't understand what this residual represents.

Thank you in advance for your help!

Shown here is my residual distribution using model.predict(operator=pde) and my code for trying to exclude boundary intersections:
![94751cd6690b447acb71dc1a683afcb](https://github.com/lululxvi/deepxde/assets/140362283/e70a7863-bfed-47bc-9dac-de0e20124c4a)
![7ec47722d7679967cc487b7e12acf50](https://github.com/lululxvi/deepxde/assets/140362283/beedb980-d2d4-4195-9715-b886b2ea1799)
![46279aa55e0801a184e78fb0c544f99](https://github.com/lululxvi/deepxde/assets/140362283/d2c33a03-4aa0-421c-b8f3-4e97fba69417)

**Comment by lululxvi** [comment]
The usage of `exclusions` is the correct way.
